src/charm.py

```diff
@@ -67,6 +97,9 @@ def _on_postgresql_pebble_ready(self, event: WorkloadEvent) -> None:
                 # Changes were made, add the new layer.
                 container.add_layer(self._postgresql_service, new_layer, combine=True)
                 logging.info("Added updated layer 'postgresql' to Pebble plan")
+                self._patroni.render_patroni_yml_file()
+                # TODO: move this file generation to on config changed hook
+                # when adding configs to this charm.
                 # Restart it and report a new status to Juju.
                 container.restart(self._postgresql_service)
```

Comment:
Why the restart is required?

Reply:
The restart is required at this point to really start patroni/postgresql (their service was not yet present in the container until we add the new layer on L98).

Reply:
Got it. Thanks for the explainer.